oauthproxy.go

```diff
@@ -585,6 +590,24 @@ func (p *OAuthProxy) SignIn(rw http.ResponseWriter, req *http.Request) {
 	}
 }
 
+//UserInfo endpoint outputs session email in JSON format
+func (p *OAuthProxy) UserInfo(rw http.ResponseWriter, req *http.Request) {
+
+	var userInfo struct {
+		Email string `json:"email"`
+	}
+
+	session, err := p.getAuthenticatedSession(rw, req)
+	if err != nil {
+		http.Error(rw, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+	userInfo.Email = session.Email
```

Comment:
Sorry for the nit, but you could remove lines 595-599 and do this:
```suggestion
	userInfo := struct {
		Email string `json:"email"`
	}{session.Email}
```

Reply:
No sorry , on the contrary, I am open to take suggestion from an experienced person. Thanks.